packages/docusaurus-theme-classic/src/theme/DocItem/index.js

```diff
@@ -14,7 +14,75 @@ import DocPaginator from '@theme/DocPaginator';
 
 import styles from './styles.module.css';
 
+let headersCache = [];
```

Comment:
These variables shouldn't be module-scoped although we most likely will not need more than one on a page. They should be scoped to each instance, either as state or refs. Refs are preferred since we are not using React to manage the rendering here.

Reply:
Are there perf implications with not caching these elements? Would prefer not to cache them if possible in case new items are added dynamically to the page, the logic will still work.

Reply:
I going to change the approach and not cache anymore, in this way we can consider dynamically added elements.
Is going to be up in the next commit.